aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -106,6 +108,50 @@ Tensor isinf(const Tensor &self) {
   });
 }
 
+static inline TensorIterator build_is_infinity_op_iterator(Tensor& result, const Tensor& self) {
+  return at::TensorIteratorConfig()
+    .check_all_same_dtype(false)
+    .set_check_mem_overlap(true)
+    .add_output(result)
+    .add_input(self)
+    .cast_common_dtype_to_outputs(true)
+    .build();
+}
+
+Tensor isposinf(const Tensor &self) {
```

Comment:
What about explicitly erroring on complex types ("isposinf does not support complex inputs"), returning `torch.full_like(self, False)` for integer (and bool) tensors and then acquiring infinity and testing for it using equality for float tensors? 

For bfloat16 you have two options. You can add a special case to error, or you can support it by acquiring its infinity from:

https://github.com/pytorch/pytorch/blob/1a0b95e7e46a056546a49f7f8b40aaf6ffa11dde/c10/util/BFloat16-inl.h#L249

Reply:
`TORCH_CHECK(!self.is_complex(), message)` is added for erroring on complex types, and the latest commit adds support for `bfloat16`.
Meanwhile, the test code has been updated for `complex` and `bfloat16` types.